server/index.ts

```diff
@@ -31,33 +30,38 @@ app.use(cookieParser());
 app.use(compression());
 app.use(bodyParser.json());
 
+if (process.env.NODE_ENV?.trim() === 'production') {
+    app.use(express.static(path.join(__dirname, '../../build')));
+} else {
+    app.use(express.static(path.join(__dirname, '../client/')));
+}
+
 // localhost:3000/gql -> graphQL sandbox
 server.start().then((): void => {
     server.applyMiddleware({ app, path: '/gql' });
 
-    // routers
-    app.use('/api/users', userRouter);
-    app.use('/auth', authRouter);
-
-    // for testing purposes
-    app.get('/api/projects', async (req, res) => {
-        const projects = await ProjectDB.find();
-        return res.json(projects);
-    });
-
     // for logger to cross reference project DB api key to request auth header
     app.get('/auth/:projectID', async (req, res) => {
-        const project = await ProjectDB.findById(req.params.projectID).catch(
+        const projectResult = await ProjectDB.findById(req.params.projectID).catch(
             (err) => new Error(`Project not found: ${err}`)
         );
-        return res.json(project.apiKey);
+
+        let apiKey = null;
+
+        if (projectResult && !(projectResult instanceof Error)) apiKey = projectResult.apiKey;
+
+        return res.json(apiKey);
     });
 
     // serve homepage
     app.all('/', (req, res) =>
         res
             .setHeader('Content-Type', 'text/html')
-            .sendFile(path.join(__dirname, '../public/index.html'))
+            .sendFile(
+                process.env.NODE_ENV?.trim() === 'production'
+                    ? path.join(__dirname, '../../public/index.html')
+                    : path.join(__dirname, '../public/index.html')
```

Comment:
Is this a mistake: `'../public/index.html'` vs .`'./../public/index.html'`

Reply:
no. when in development, the server is in `./server`, when in production, the server is in `./ts-build/server`